pyrelic/__init__.py

```diff
@@ -48,6 +48,9 @@
     # pairings
     pair,
     pair_product,
+
+    on_curve_G2,
+    on_curve_G1,
```

Comment:
I'm not really sure whether these two functions should be exposed publicly. It would be nicer to be sure that if you have an instance of `G1` or `G2`, that the points are definitely on the curve. The only place where one could currently create instances that are not on the curve are their constructors with a non-`None` byte buffer. Checking after deserialization there and throwing a `ValueError` would make sense. If any of the other functions are known to produce incorrect results (such as `rand_Gx` or `hash_to_Gx`), then they either should retry (`rand_Gx`) or raise an exception (`hash_to_Gx`).

Reply:
the only case that Gx get wrongly generated will lead to an error `: ERROR in ep2_read_bin() at /Users/xxxx/Project/relic/src/epx/relic_ep2_util.c,252: insufficient buffer capacity.`, and return the inifinity point .  So, I think we don't need `on_curve_Gx`  at all?

Reply:
Yeah, that's a part that I'm not really sure yet on the best way to handle it. Could you split off the test case without the `on_curve_Gx` parts for now?

Reply:
sure.  plz recheck if it makes sense to add a new unittest.  Thanks!

Reply:
@duanbing Thanks, merged (with a small change). I've also pushed checks for `*_read_bin` failures.